wallet/src/account/mod.rs

```diff
@@ -2132,7 +2308,8 @@ fn group_preselected_inputs(
     chain_config: &ChainConfig,
     block_height: BlockHeight,
     dest_info_provider: Option<&dyn DestinationInfoProvider>,
-) -> Result<BTreeMap<currency_grouper::Currency, (Amount, Amount)>, WalletError> {
+    order_info: Option<&RpcOrderInfo>,
```

Comment:
This should be a map <OrderId, &RpcOrderInfo> just in case to know that the order info is for the order from the inputs.